demo/run-osm-demo.sh

```diff
@@ -30,6 +30,7 @@ IMAGE_PULL_POLICY="${IMAGE_PULL_POLICY:-Always}"
 ENABLE_EGRESS="${ENABLE_EGRESS:-false}"
 MESH_CIDR=$(./scripts/get_mesh_cidr.sh)
 DEPLOY_WITH_SAME_SA="${DEPLOY_WITH_SAME_SA:-false}"
+ENVOY_LOG_LEVEL="${ENVOY_LOG_LEVEL:-debug}"
```

Comment:
nice

Reply:
Should the default be error?

Reply:
For the local development runs and the CI no, so that we can override the defaults in the install cli. Otherwise it would make it hard to debug CI failures. 
We could set it to `error` here and override using env variables, but most use cases of the automated demo script are from CI and local development runs.